### PR TITLE
Fix `dotnet sln add` creating empty solution folders

### LINKED ISSUE #11793: CLI issue with "dotnet sln add -s"
When using `dotnet sln add -s` to add a project to non-existing solution folder, an extra `Project` line with empty quotes is added to the .sln. This causes the .sln to appear corrupted. Removal of the line fixes things. Adding to an existing solution folder doesn't cause the issue.

This was done through cmd; not tested in PowerShell.

Repro:
1. Use any solution.
2. From the solution folder, create a project via cli, such as `dotnet new classlib -f netcoreapp3.1 -n Dummy -o Dummy`.
3. Add the new project to the solution, and put it in a solution folder (that doesn't yet exist) using `-s`. IE: `dotnet sln add -s \Dummy Dummy\Dummy.csproj`
4. Open or return to Visual Studio, and you'll see an exception for a corrupted .sln.

If you look at the .sln, you'll find a couplet like this:
```
Project("{guid}") = "", "", "{guid}"
EndProject
```

Removing that fixes the problem.

### PR DESCRIPTION
**Pull request by ZeroErrors**
Fixes https://github.com/dotnet/sdk/issues/11793

## Change log
- Added `StringSplitOptions.RemoveEmptyEntries` when handling `SlnAddParser.SolutionFolderOption` to prevent empty solution folders.
- Normalize directory seperator in `SlnAddParser.SolutionFolderOption` so either `Path.DirectorySeparatorChar` or `Path.AltDirectorySeparatorChar` can be used.
- Added unit tests for above cases
- Removed unused private fields in `AddProjectToSolutionCommand`

## Description
When using the `dotnet sln add` command with a provided solution folder is was possible for the command to add empty solution folders if the provided path contained multiple simultaneous directory separators or had a directory separator as a prefix or suffix.

To prevent this from happening the `StringSplitOptions.RemoveEmptyEntries` was added to the `string.Split(...)` method call when getting the list of folders in the path. This prevents any empty strings in the list and so prevents empty solution folders from being created.

The command also failed to recognize the alt directory separator allowing solution folders to contain `/` characters on windows.
To solve that problem the path's directory separators are now normalized to be `Path.DirectorySeparatorChar` so the path can be correctly split.

## Testing
Unit tests for the above problems included.

To test manually setup a solution and a project with
```
dotnet new sln
dotnet new console --output src/todo-app
```

**Windows**
Run one of the following:
`dotnet sln add src/todo-app --solution-folder todo`
`dotnet sln add src/todo-app --solution-folder /todo//`
`dotnet sln add src/todo-app --solution-folder \todo\\`

**Linux**
Run one of the following:
`dotnet sln add src/todo-app --solution-folder todo`
`dotnet sln add src/todo-app --solution-folder /todo//`

All of these now add `src/todo-app` in the solution folder `todo`.
```sln
Project("{GUID}") = "todo", "todo", "{GUID}"
EndProject
Project("{GUID}") = "todo-app", "src\todo-app\todo-app.csproj", "{GUID}"
EndProject
```

<details>
 <summary>The previous functionality (main)</summary>

`dotnet sln add src/todo-app --solution-folder /todo//`
```sln
Project("{GUID}") = "/todo//", "todo\", "{GUID}"
EndProject
Project("{GUID}") = "todo-app", "src\todo-app\todo-app.csproj", "{GUID}"
EndProject
```
_Note: On linux this produces the output of the command below below._
<br/>

`dotnet sln add src/todo-app --solution-folder \todo\\`
```sln
Project("{GUID}") = "", "", "{GUID}"
EndProject
Project("{GUID}") = "todo", "todo", "{GUID}"
EndProject
Project("{GUID}") = "", "", "{GUID}"
EndProject
Project("{GUID}") = "", "", "{GUID}"
EndProject
Project("{GUID}") = "todo-app", "src\todo-app\todo-app.csproj", "{GUID}"
EndProject
```
<br/>

`dotnet sln add src/todo-app --solution-folder todo`
```sln
Project("{GUID}") = "todo", "todo", "{GUID}"
EndProject
Project("{GUID}") = "todo-app", "src\todo-app\todo-app.csproj", "{GUID}"
EndProject
```
</details>

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=18398) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=18398) before we can accept your contribution.<br/><br/>:x: ZeroErrors [sign now](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=18398)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/sdk?pullRequest=18398) it.</sub>

File: src/Tests/dotnet-sln.Tests/GivenDotnetSlnAdd.cs [hunk 1]
```diff
@@ -1186,6 +1186,48 @@ public void WhenSolutionFolderAndInRootIsPassedItFails()
                 .Should()
                 .BeVisuallyEquivalentTo(contentBefore);
         }
+        
+        [Fact]
+        public void WhenSolutionFolderIsPassedWithDirectorySeparatorFolderStructureIsCorrect()
```
Comment (danmoseley) [inline src/Tests/dotnet-sln.Tests/GivenDotnetSlnAdd.cs:1191]: can you collapse these tests into one by parameterizing using `[InlineData..]` ?

**Comment by danmoseley** [comment]
Thanks for asking but someone in @marcpopMSFT  team will have to review this, I don't have familiarity.

**Review by marcpopMSFT — APPROVED** [review]

**Comment by marcpopMSFT** [comment]
Thank you for your contribution!

**Pull request merged**